educator/src/Dscp/Educator/Web/Student/Auth.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE DataKinds     #-}
+-- | Nessesary types and implementation for Student authenthication
+
+module Dscp.Educator.Web.Student.Auth
+       ( StudentAuth
+       , WithStudent (..)
+       , AuthData (..)
+       , GetStudents
+       ) where
+
+import Crypto.JOSE.JWK hiding (hash)
+import Crypto.JWT hiding (hash)
+import Data.Aeson
+import Servant.Auth.Server
+import Servant.Auth.Server.Internal.Class
+
+import Dscp.Crypto (PublicKey, AbstractPK (..), hash)
+import Dscp.Core (Address (..), Student)
+
+---------------------------------------------------------------------------
+-- Data types
+---------------------------------------------------------------------------
+
+-- | Custom authenthication type
+data StudentAuth
+
+-- | A type that has a Student attached to it
+data WithStudent a = WithStudent Student a
+
+-- | A type that the student has sent to authenthicate
+data AuthData = AuthData
+  deriving Generic
+
+-- | Action to get all students
+type GetStudents = IO [PublicKey]
```

Comment:
newtype

Reply:
This type represents an action to get the list of students (or rather, their public keys). That's why I thought `GetStudents` name is fitting. I can make it `GetStudentsAction` I guess.

Reply:
I don't really see a difference between `IO` action and any other value, why is it a big deal to not have it behind a type?

I'm using an `IO` action here, instead of something like `TVar [PublicKey]` because that allows the auth checker to not depend or even care about the implementation details on how it is actually going to get the public keys (is it TVar? Is it a database connection? etc), it just executes the action and gets the thing. All implementation details are left to whoever provides the servant context.

Reply:
Yeah, it will be a `newtype` after I update and finalize the PR.